### DISCUSSION TITLE
Users not automatically assigned to groups using LDAP/AD authentication

### DISCUSSION THREAD

**Discussion by githubdave2020** [requarks/wiki#3719] [Error / Bug Report]
**Describe the bug**
We are using LDAP / Active Directory authentication and have allowed "Self Registration" and entered a valid group for **new** users to be added to upon authentication.
We have adjusted the GUEST access to require users to be authenticated to view any items, therefore, new users that authenticate are successful and we see them as users, however, they do not get automatically assigned to the group we have specified.  These user's experience a successful login and a redirect back to the login page.

**To Reproduce**
1. Set up and confirm LDAP/AD authentication is working properly.
2. Enable "Allow Self Registration"
3. Enter a valid group in the "Assign to Group" section
4. Ensure the group assigned in # 3 above can view all pages.
5. Ensure users within the group can authenticate and navigate all pages.
6. Find a user in the LDAP/AD that has not registered.  
7. Enter their credentials in the login page.
8. The user will be authenticated and will show up in the users section, but will not have the default group assigned to them.
9. If the GUEST access group has no permissions to view, the user will be redirected back to the login page.

**Expected behavior**
New users that authenticate with AD should be placed in the default group that has been selected in the self-registration "Assign new users to this group"  section.

We have enabled LDAP logging, however, there was no output of value to determine the issue

**Host Info (please complete the following information):**
 - OS: Linux - Red Hat Linux (maipo) 7.7 x64
 - Wiki.js version: 2.0.12
 - Database engine: MySQL 8.0.19

**Comment by Anishmourya** [comment]
@NGPixel  This is happening in SAML 2.0 authentication as well, however user created successfully and added to users db table as well but userGroups doesn't have any row for user. 

      // Assign to group(s)
      if (provider.autoEnrollGroups.length > 0) {
        await user.$relatedQuery('groups').relate(provider.autoEnrollGroups)
      }
line number 263 users.js in models . while  autoEnrollGroups have {"v":[4,2]} value in db .
i did manually update autoEnrollGroups value in authentication table from db . then it works 
seems like UI not updating values some times (Allow Self Registration) toggle button may not sending value to server. 
btw its working fine for me. cheers
![Screenshot 2020-09-11 at 6 08 40 PM](https://user-images.githubusercontent.com/6560320/92926875-521ba980-f45a-11ea-8a16-789a7d5d65fa.png)
